scipy/io/mmio.py

```diff
@@ -317,8 +373,23 @@ def __init__(self, **kwargs):
         self._init_attrs(**kwargs)
 
     #---------------------------------------------------------------------------
-    def read(self, source):
-        stream, close_it = self._open(source)
+    def read(self, filename):
+        """
+        Reads the contents of a Matrix Market file 'filename' into a matrix.
+    
+        Parameters
+        ----------
+    
+        filename : file
```

Comment:
Yes, it would be better to rename it. One question: Is there a difference between a `file` object and a `file-like` object? https://docs.python.org/3/glossary.html#term-file-object states that these are synonyms.